docs/naps/7-key-binding-dispatch.md

```diff
@@ -0,0 +1,427 @@
+(nap-7)=
+
+# NAP-7 — Key Binding Dispatch
+
+```{eval-rst}
+:Author: "Kira Evans <mailto:contact@kne42.me>"
+:Created: 2023-07-03
+:Status: Draft
+:Type: Standards Track
+```
+
+## Abstract
+
+With the switching of the internal key binding system to use app-model's representation[^id1], there is discussion as to what exactly constitutes a valid key binding and how conflicts are handled[^id2].
+
+This NAP seeks to clarify and propose a solution for how key bindings will be dispatched according to their priority, enablement, and potential conflicts.
+
+## Motivation and Scope
+
+Plugin developers are able to export commands in their manifest file but cannot similarly set their shortcuts in a code-free way. npe2 provides an option to bind commands to key binding but it is undocumented and unsupported since napari still uses an old dispatch system of chainmaps.
+
+### A more versatile system
+
+The proposed dispatch system would leverage weights to clearly separate default, plugin, and user-defined key binding as well as support a more advanced conditional system which would determine if a keybinding is active/enabled or not. Both of these properties are already part of the specification defined by app-model.
+
+Separation of different sources of key binding makes it easier for a user to determine who set the binding as well as how to restore the default.
+
+Conditional evaluation allows plugin developers to tie their keybinding to a specific viewer state.
+
+### Definitions
+
+**modifier keys** refer to `ctrl`, `shift`, `alt`, and `meta`. `meta` is also known as `cmd`, `win`, or `super` on osx, windows, or linux, respectively
+
+a **base key** is one of the following supported keys:
+- `f1-f12`, `a-z`, `0-9`
+- `` ` ``, `-`, `=`, `[`, `]`, `\`, `;`, `'`, `,`, `.`, `/`
+- `left`, `up`, `right`, `down`, `pageup`, `pagedown`, `end`, `home`
+- `tab`, `enter`, `escape`, `space`, `backspace`, `delete`
+- `pausebreak`, `capslock`, `insert`, `numlock`, `printscreen`
+- `numpad0-numpad9`, `numpad_decimal`, `numpad_multiply`, `numpad_divide`, `numpad_add`, `numpad_subtract`
+
+a **key combination** is a base key pressed with one or more modifier keys, e.g. `ctrl+c` or `ctrl+shift+z`
+
+a **key chord** consists of two parts, of which each can be either a base key or a key combination, e.g, `ctrl+x v`
```

Comment:
What do you mean by two parts? A key combo represent simultaneously pressing several keys, but a chord a sequence? Must each part be pressed separately?

Reply:
yes, each part needs to be pressed separately. they do not need to be within a certain time gap, although that was a consideration for indirect conflicts (e.g. `ctrl+x` is both its own kb and also part of the chord `ctrl+x m`)